### PR TITLE
Add symbol / string versions of find_library

### PR DESCRIPTION
**Pull request by jakebolewski**
Relax `find_library` argument types to any iterable

### CODE REVIEW

File: test/libdl.jl [hunk 1]
```diff
@@ -22,7 +22,9 @@ end
 
 cd(dirname(@__FILE__)) do
 
-# @test !isempty(Libdl.find_library(["libccalltest"], [dirname(@__FILE__)]))
+@test !isempty(Libdl.find_library(["libccalltest"], [dirname(@__FILE__)]))
```
Comment (tkelman) [inline test/libdl.jl]: this no longer builds in `test/`, it's under `$(build_shlibdir)` now

Comment (jakebolewski) [inline test/libdl.jl]:
```shell
Julia/julia-dev [jcb/findlib●] » ls -l test/* | grep ccalltest
-rwxr-xr-x  1 jacobbolewski  staff  16528 Sep 13 15:25 test/libccalltest.dylib
test/libccalltest.dylib.dSYM:
``` 

?

Comment (tkelman) [inline test/libdl.jl]: that's stale and leftover from when it used to build there

**Comment by tkelman** [comment]
lgtm

**Pull request merged**